source/utf8/unchecked.h

```diff
@@ -276,62 +276,73 @@ namespace utf8
             }
           }; // class bidirectional_iterator
 
-        template <typename octet_iterator>
-        class input_iterator : public std::iterator<std::input_iterator_tag, uint32_t> {
-          private:
-          octet_iterator it;
-          uint32_t cp{};
-          void read() {
-              cp = utf8::unchecked::next_impl(it);
-          }
-    
+        class iterator_stream {
```

Comment:
Why not actually call  `istream_iterator`?

Reply:
I don't understand. I want an iterator with value_type `uint32_t`.

Reply:
Sure. This `utf8::iterator_stream` is a wrapper over a `std::istream_iterator<char>`, so why not use call it `utf8::istream_iterator` instead? Just sounds more consistent instead of swapping the words

Reply:
sounds good.